internal/service/streamprocessor/model.go

```diff
@@ -51,3 +111,42 @@ func convertStatsToTF(stats any) (types.String, diag.Diagnostics) {
 	}
 	return types.StringValue(string(statsJSON)), nil
 }
+
+func extractChangeStreamTokenFromStats(stats any) (types.String, diag.Diagnostics) {
+	if stats == nil {
+		return types.StringNull(), nil
+	}
+	var statsMap map[string]interface{}
+
+	statsJSON, err := json.Marshal(stats)
+	if err != nil {
+		return types.StringValue(""), diag.Diagnostics{diag.NewErrorDiagnostic("failed to marshal stats", err.Error())}
+	}
+
+	err = json.Unmarshal(statsJSON, &statsMap)
+	if err != nil {
+		return types.StringValue(""), diag.Diagnostics{diag.NewErrorDiagnostic("failed to unmarshal stats", err.Error())}
+	}
+
```

Comment:
Maybe we can instead cast directly to map[string]interface ? Not sure marshalling and unmarshalling is necessary